### PR TITLE
Please do not include Chinese in commit and PR message

### PR DESCRIPTION
**Pull request by creasy2010**
流程对下. 主要看下createFile是否有差异 ..

### CODE REVIEW

File: src/handlers/createFile.ts [hunk 1]
```diff
@@ -0,0 +1,43 @@
+import * as fs from 'fs';
+import * as mkdirp from 'mkdirp';
+import * as path from 'path';
+import * as ejs from 'ejs';
+import {isUndefined} from 'lodash';
```
Comment (zhangkaiyulw) [inline src/handlers/createFile.ts]: please use !isDefined and remove loadash, please make sure { always space around }

File: src/handlers/appendFile.ts [hunk 2]
```diff
@@ -6,11 +6,11 @@ import Reporter from '../Reporter';
 import AppendFileInfo from '../instructions/AppendFileInfo';
 import Render from '../Render';
 
-const appendFile = (
+export default function appendFile  (
```
Comment (zhangkaiyulw) [inline src/handlers/appendFile.ts]: please change back to const blabla and export default blabla

File: src/handlers/createFile.ts [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+import CreateFileInfo from '../instructions/CreateFileInfo';
+
+export default function createFile({
+  content,
```
Comment (zhangkaiyulw) [inline src/handlers/createFile.ts]: please name this parameter params and add render and reporter like how `appendFile` does.

File: src/handlers/createFile.ts [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+  }
+  if (context) {
+    content = ejs.render(content, context);
+  }
+  const dest = at;
+  if (fs.existsSync(dest)) {
+    const destContent = fs.readFileSync(dest).toString();
+    if (destContent === content) {
+      // never mind, same content
+      return ['up-to-date', 'yellow', at];
```
Comment (zhangkaiyulw) [inline src/handlers/createFile.ts]: please use reporter api

File: src/handlers/createFile.ts [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+      } else {
+        // jsut errors
+        return ['exist', 'red', at];
+      }
+    }
+  } else {
+    // create file
+    mkdirp.sync(path.dirname(dest));
+    fs.writeFileSync(dest, content);
+    return ['create', 'green', at];
```
Comment (zhangkaiyulw) [inline src/handlers/createFile.ts]: please use reporter api and remove color argument

File: src/handlers/deleteFile.ts [hunk 6]
```diff
@@ -0,0 +1,16 @@
+import * as fs from 'fs';;
+import * as path from 'path';
+
+export default function deleteFile ({ at, silent }) {
```
Comment (zhangkaiyulw) [inline src/handlers/deleteFile.ts]: please use 2 arguments, params and reporter

File: src/handlers/detachFromFile.ts [hunk 7]
```diff
@@ -0,0 +1,29 @@
+import * as fs from 'fs';;
+import * as path from 'path';
+import * as ejs from 'ejs';
+import {isUndefined,endsWith} from  'lodash';
```
Comment (zhangkaiyulw) [inline src/handlers/detachFromFile.ts]: please remove isUndefined and endsWith, use !isDefined and string.endsWith instead

File: src/handlers/index.ts [hunk 8]
```diff
@@ -0,0 +1,13 @@
+export {default as createFile} from  './createFile';
```
Comment (zhangkaiyulw) [inline src/handlers/index.ts]: Very smart implementation! 👍👍

File: src/handlers/createFile.ts (hunk 5 above)
Comment (creasy2010) [reply src/handlers/createFile.ts]: I understand a little.

**Pull request closed without merging**